### PR TITLE
ios: fix platform key value store

### PR DESCRIPTION
**Pull request by Augustyniak**
Commit Message:
Additional Description: Fixes a typo kind of bug in the implementation of the iOS platform key value store and adds a test coverage for iOS platform key value store to test other parts of the implementation and avoid regressions.
Risk Level: Low
Testing: Unit tests
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: mobile/library/objective-c/EnvoyEngineImpl.m [hunk 1]
```diff
@@ -403,8 +403,8 @@ static void ios_kv_store_save(envoy_data native_key, envoy_data native_value, co
     NSString *key = [[NSString alloc] initWithBytes:native_key.bytes
                                              length:native_key.length
                                            encoding:NSUTF8StringEncoding];
-    NSString *value = [[NSString alloc] initWithBytes:native_key.bytes
-                                               length:native_key.length
+    NSString *value = [[NSString alloc] initWithBytes:native_value.bytes
```
Comment (Augustyniak) [inline mobile/library/objective-c/EnvoyEngineImpl.m:406]: this is the actual bug fix.

Comment (abeyad) [reply mobile/library/objective-c/EnvoyEngineImpl.m:406]: good catch!  looks like it was a copy-paste bug.

**Review by abeyad — APPROVED** [review]
Thanks for adding the test Rafal!  I don't know Objective-C, but the fix and the contents of the test LGTM.

File: mobile/library/objective-c/BUILD [hunk 2]
```diff
@@ -49,6 +47,8 @@ objc_library(
     }),
     visibility = ["//visibility:public"],
     deps = [
+        ":envoy_aliases_lib",
```
Comment (jpsim) [inline mobile/library/objective-c/BUILD:50]: Doesn't this dep need to go in `envoy_objc_bridge_lib` so that the header is available when building it?

Comment (Augustyniak) [reply mobile/library/objective-c/BUILD:50]: it's a part of `envoy_objc_bridge_lib ` already - but you are right, it's redundant here I think. Removing

**Review by jpsim — APPROVED** [review]

**Pull request merged**